Lib/unittest/test/test_case.py

```diff
@@ -1839,5 +1839,24 @@ def test2(self):
             self.assertEqual(MyException.ninstance, 0)
 
 
+class AssertEqualTest(unittest.TestCase):
+
+    def test_trailing_new_line_at_end(self):
+        self.assertEqual("abc\n", "abc\n")
```

Comment:
I think a better test would be to test the output diff like https://github.com/python/cpython/blob/e709c1be97a04156c628f534116a236b9581aa2f/Lib/unittest/test/test_case.py#L1122 and it should be testing the failure cases and asserting the Exception output string.